lib/src/anchor.dart

```diff
@@ -21,4 +21,27 @@ class Anchor {
   Vector2 translate(Vector2 p, Vector2 size) {
     return p - (toVector2..multiply(size));
   }
+
+  static final Map<Anchor, String> _valueNames = {
```

Comment:
Isn't the current way more like an enum than this way of parsing strings?
What is the thought behind adding this second way of getting an Anchor?

Reply:
we discussed offline, the key reason here is serialization

> re about the anchor class: let me explain
you are questioning why I added methods to convert to and from string and whether that is an antipattern. those methods are to make serialization easy. enums not only have a values list but they are intrinsically serializable. objects are not. if you want to store something that has an anchor as json, for example, you would need to write something like I did yourself. in fact to use them with dashbook erick had to do that by hand. so I believe having these methods makes it easier to do so if you ever wanted, and the code is in one centralized place. enums in java also offer .name and .valueOf for that purpose so I think it's quite expected that enums have to/from string conversion (maybe I can use the java names to make it clear). I feel being Serializable in a sense (even though the concept is not as precisely defined as in java/kt) is important.

Reply:
Is telegram offline? 👀